### PR TITLE
feat(brand-claims): brand-claims audit (enable + run) for scheduled Brand Claims trigger

### PR DESCRIPTION
**Pull request by akshaymagapu**
## What

Adds a `brand-claims` audit that, for a given `siteId`, does both steps needed to trigger a Brand Claims run, in one place:

1. **Enable** — sets `brand_claims_enabled = true` on the site's active brand (idempotent: skips the write if already on; skips the run if the brand was deleted mid-flight).
2. **Run** — discovers the brand's latest Brand Presence sheet in S3 (`DRS_BP_BUCKET`) and publishes the DRS `BRAND_PRESENCE_SHEET_WRITTEN` event to `SQS_BP_SHEET_READY_QUEUE_URL`.

Registered as the `brand-claims` audit type (a plain SQS handler, not an AuditBuilder audit). It mirrors the api-service `enable-brand-claims` + `run-brand-claims` Slack commands, combined.

## Why this shape

This is the lean alternative to standing up a dedicated DRS provider for the same goal: it reuses the audit worker's existing SQS dispatch and scheduling, so there is no new queue/CDK/provider-framework surface. Scheduling is a config change (see below), not new infrastructure.

## Scheduling (follow-up, separate change)

The audit is scheduled weekly per-site via the existing mechanism — a `spacecat-jobs-dispatcher` cron entry for the `brand-claims` type + enabling the handler for the target sites in the Configuration. No code change here beyond the audit itself.

## Env (ops)

Reads from `context.env`, to be added in Vault:
- `SQS_BP_SHEET_READY_QUEUE_URL`
- `DRS_BP_BUCKET`

## Review fixes already applied

- Infra/config faults (missing queue URL / bucket / postgrest client) **throw** so SQS retries + DLQs rather than silently dropping the message.
- Enable no-op row (brand deleted mid-flight) → warn and skip the run instead of logging false success.
- S3 prefix + event use the canonical `site.getId()`, not the raw message `siteId`.
- Warn when the S3 listing hits its page cap (guards against selecting a stale sheet).
- Tests assert the brands write payload + query filters, the full published event, and `sanitizePathComponent` golden vectors.

## Testing

24 unit tests, 100% line/branch/statement coverage on the new handler; lint clean.

Docs: `docs/decisions/003-brand-claims-audit-combines-enable-and-run.md`.

Epic: LLMO-5741 (Brand Claims: GA Readiness). Note: a heavier DRS-provider implementation of the same goal exists at adobe-rnd/llmo-data-retrieval-service#2910 and is parked in favor of this leaner approach.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Change Management

```yaml
cm-assessment: v1
changeType: standard
impact: unnoticeable
risk: minor
changeApprovedBy: ["Dominique Jäggi"]
rationale: "Auto-added baseline (no PR assessment present at CI time); refine via the cmr skill if this change is higher-impact."
```

### CODE REVIEW

File: docs/decisions/003-brand-claims-audit-combines-enable-and-run.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# 003 — Brand Claims Audit: Combine Enable + Run in the Audit Worker
```
Comment (MysticatBot) [inline docs/decisions/003-brand-claims-audit-combines-enable-and-run.md]: issue (blocking): **Missing spec in `docs/specs/`.**

The repo CLAUDE.md states:
> "Significant new functionality must be developed spec-first. Add a spec to `/docs/specs/` following the template."

This PR introduces a new audit type with new integrations (PostgREST brands write, S3 listing, SQS publish to DRS), a new write path to a table owned by another service, and duplicated cross-service logic. The `docs/specs/` directory has 12+ specs for similar-scope features.

The ADR (003) documents the decisions well but an ADR is not a spec - it covers "what we chose and why" rather than "problem statement, goals, full technical design, alternatives, success criteria."

**Options:**
1. Add a spec to `docs/specs/` (e.g., `2026-08-11-brand-claims-scheduled-trigger.md`) covering the full feature design.
2. If the ADR is intended to serve as the spec, update it to include the missing spec sections (problem statement, goals, success criteria) and add an exemption note in the PR body explaining why a separate spec is redundant.

Note: the ADR + PR body together are thorough - this is a letter-of-the-rule gap, not a missing-rationale gap.

File: src/brand-claims/handler.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
+/*
```
Comment (MysticatBot) [inline src/brand-claims/handler.js:1]: issue (blocking): **CLAUDE.md "Adding a New Audit" section is now stale.**

The repo CLAUDE.md documents:
> "1. Create `src/[audit-name]/handler.js` with `AuditBuilder` and register it in `src/index.js` under `HANDLERS`."

This handler is a plain async function, not an AuditBuilder audit. While there are precedents for non-AuditBuilder handlers in the repo (`offsite-brand-presence-drs-status`, `rum-config-refresh`, `dummy`), the CLAUDE.md does not document this exception path.

**Fix:** Update the CLAUDE.md "Adding a New Audit" section to note the exception:
> "When the handler is not a traditional site audit (e.g., it performs an operational trigger rather than auditing a URL), it may be registered as a plain async function without AuditBuilder. Examples: `brand-claims`, `rum-config-refresh`."

This prevents future contributors from assuming all entries in HANDLERS must use AuditBuilder.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @akshaymagapu,

**Verdict: Request changes** - two documented convention gaps to resolve before merge.
**Complexity: HIGH** - large diff (682 lines, new audit handler + ADR + tests).
_Changes: Adds a `brand-claims` audit handler that enables the brand claims gate and publishes the DRS ready-signal for a site's latest Brand Presence sheet (4 files)._
Note: Recommend a human read before merge - this change adds an architectural decision record in `docs/decisions/`. The bot review is a complement to, not a replacement for, a human read here.

## Must fix before merge

1. [Important] Missing spec in `docs/specs/` - CLAUDE.md requires spec-first development for significant new functionality - `docs/decisions/003-brand-claims-audit-combines-enable-and-run.md:1` (details inline)
2. [Important] CLAUDE.md "Adding a New Audit" section is now stale - states all audits use `AuditBuilder`, but this handler does not - `src/brand-claims/handler.js:1` (details inline)

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- nit: ADR status should be `Accepted` at merge time, not `Proposed` (per CLAUDE.md status values) - `docs/decisions/003-brand-claims-audit-combines-enable-and-run.md:3`
- nit: Missing `siteId` logs at ERROR then returns `ok()` - inconsistent with the ADR's failure policy (infra faults throw, business no-ops warn+ack). Either downgrade to `log.warn` (terminal no-op) or throw (system fault for DLQ investigation) - `src/brand-claims/handler.js:128`
- suggestion: Add `.limit(2)` to the `getBrandForSite` query to make the "we only use the first result" intent explicit and guard against unbounded result sets - `src/brand-claims/handler.js:50`

</details>

Note: CI checks are currently pending - confirm build passes before merge.

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 4m 18s | <b>Cost:</b> $5.19 | <b>Commit:</b> <code>854fd201ea233b7d60aa816894e549644eee025a</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by akshaymagapu** [comment]
Thanks for the review. On the two blocking items:

This isn't significant new functionality — it's a small audit change: a single plain operational handler that enables the gate and re-publishes an existing ready-signal. It follows the established plain-handler precedent already in the repo (`rum-config-refresh`, `offsite-brand-presence-drs-status`, `dummy`), which don't use `AuditBuilder` and didn't require spec-first docs.

**Comment by akshaymagapu** [comment]
Addressed the review comments:
- CLAUDE.md "Adding a New Audit": documented the plain operational-handler exception (brand-claims, rum-config-refresh, offsite-brand-presence-drs-status).
- Missing `siteId` now `log.warn` + ok() (terminal no-op), not error.
- Added `.limit(2)` to the brand query.
- Spec/ADR: this is a small audit-handler addition (not significant new functionality), following existing plain-handler precedent — ADR removed; no separate spec added.

File: src/brand-claims/handler.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (MysticatBot) [inline src/brand-claims/handler.js:1]: issue (blocking): **Missing spec.** CLAUDE.md states: "Significant new functionality must be developed spec-first. Add a spec to `/docs/specs/`." This handler is 236 lines of new multi-system integration (PostgREST, S3, SQS) implementing a novel non-AuditBuilder pattern. The original ADR was removed in the latest commit, leaving no design document at all.

The spec does not need to be heavy - a one-page doc covering trigger, event contract, failure modes, and the rationale for bypassing AuditBuilder would satisfy the requirement and give the next maintainer a single place to understand the cross-system coupling.

File: test/audits/brand-claims.test.js [hunk 4]
```diff
@@ -0,0 +1,384 @@
+/*
```
Comment (MysticatBot) [inline test/audits/brand-claims.test.js:1]: issue (blocking): **No test for S3/SQS failure propagation.** The handler correctly lets infrastructure errors propagate (no try/catch around `s3Client.send` or `sqs.sendMessage`), which triggers SQS retry and eventual DLQ delivery. But no test asserts this contract. If a future refactor wraps these calls in try/catch with warn+ok (a pattern used elsewhere in this handler for business no-ops), messages would be silently lost.

Suggested additions (two small tests):
```js
it('throws (SQS retry) when S3 listing fails', async () => {
  s3Client.send.rejects(new Error('AccessDenied'));
  await expect(brandClaimsHandler(message, buildContext()))
    .to.be.rejectedWith('AccessDenied');
});

it('throws (SQS retry) when SQS publish fails', async () => {
  s3Client.send.resolves({
    Contents: [{ Key: `${SITE_ID}/acmecorp/analytics/chatgpt_free/2026/01/01/bp-w1-2026.xlsx`, LastModified: new Date() }],
    IsTruncated: false,
  });
  sqs.sendMessage.rejects(new Error('ServiceUnavailable'));
  await expect(brandClaimsHandler(message, buildContext()))
    .to.be.rejectedWith('ServiceUnavailable');
});
```

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @akshaymagapu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - missing spec and untested failure-propagation paths.
**Complexity: HIGH** - large diff (621 lines, new audit handler + tests, multi-system integration).
_Changes: Adds a `brand-claims` SQS handler that enables the brand claims gate on a site's active brand and publishes the DRS ready-signal for its latest Brand Presence sheet (3 files)._
Note: Recommend a human read before merge - this change adds/removes an architectural decision record in `docs/decisions/`, and the conventions reviewer flagged spec divergence (heuristic). The bot review is a complement to, not a replacement for, a human read here.
Note: CI checks are currently pending - confirm build passes before merge.

## Must fix before merge

1. [Important] Missing spec in `docs/specs/` - CLAUDE.md requires spec-first development for significant new functionality - `src/brand-claims/handler.js:1` (details inline)
2. [Important] No test for S3/SQS failure propagation - critical retry paths have no test proving errors propagate rather than being swallowed - `test/audits/brand-claims.test.js:1` (details inline)

<details>
<summary>Non-blocking (4): minor issues and suggestions</summary>

- nit: `log.error` + `return ok()` for missing `siteId` sends contradictory signals - ERROR implies actionable, but acking discards the message. WARN fits better since retrying a structurally invalid message cannot succeed - `src/brand-claims/handler.js:128`
- nit: CLAUDE.md "Adding a New Audit" section states all audits use `AuditBuilder`, which is already stale (other handlers bypass it too) - consider updating to document the plain-handler pattern - `src/brand-claims/handler.js:1`
- suggestion: Add `.limit(2)` to the `getBrandForSite` query to express intent and bound results when the invariant (one brand per site) is violated - `src/brand-claims/handler.js:50`
- suggestion: Pin the exact hash output in the `sanitizePathComponent` fallback test rather than just checking the format - guards against silent algorithm drift when this must match DRS byte-for-byte - `test/audits/brand-claims.test.js:96`

</details>

## Previously flagged, now resolved

- ADR status "Proposed" at merge time - resolved (ADR removed)

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 49s | <b>Cost:</b> $5.61 | <b>Commit:</b> <code>edf467a126759fe36312112d1424e7d7eb0f5328</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by akshaymagapu** [comment]
Added a one-page spec at `docs/specs/2026-08-11-brand-claims-scheduled-trigger.md` (problem, goals, non-goals, technical design incl. the event contract + failure policy + the non-AuditBuilder rationale, alternatives, success criteria). Also added the S3/SQS failure-propagation tests, pinned the sanitize hash, `log.warn` on missing siteId, the CLAUDE.md plain-handler note, and `.limit(2)`. Re-requesting review.

**Review by MysticatBot — APPROVED** [review]
Hey @akshaymagapu,

**Verdict: Approve** - all prior blocking findings addressed, no new concerns.
**Complexity: HIGH** - large diff (769 lines, new audit handler + spec + tests).
_Changes: Adds a `brand-claims` SQS handler that enables the brand claims gate and publishes the DRS ready-signal, with spec, CLAUDE.md update, and comprehensive tests (5 files)._
Note: Recommend a human read before merge - this change modifies root CLAUDE.md and adds a spec in `docs/specs/`. The bot review is a complement to, not a replacement for, a human read here.

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- nit: CLAUDE.md exception text lists `brand-claims` as an "Existing example" but it is introduced by this same PR - consider "Examples:" instead - `CLAUDE.md:248`

</details>

## Previously flagged, now resolved

- Spec added at `docs/specs/2026-08-11-brand-claims-scheduled-trigger.md`
- S3/SQS failure-propagation tests added (throw contract guarded)
- Missing siteId now `log.warn` + ack (was `log.error`)
- CLAUDE.md "Adding a New Audit" section updated with plain-handler exception
- `.limit(2)` added to brand query
- `sanitizePathComponent` hash fallback test pinned to exact value

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 6s | <b>Cost:</b> $5.82 | <b>Commit:</b> <code>fbe30fd07289a16cc69c334e003821fddc0497cd</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @akshaymagapu,

**Verdict: Approve** - clean incremental improvements, no new concerns.
**Complexity: HIGH** - large diff (779 lines, new audit handler + spec + tests).
_Changes: Adds Monday-partition filtering to the brand-claims sheet selector and gates the run on `brand_claims_enabled`, making the handler read-only against the brands table (5 files)._
Note: Recommend a human read before merge - this change modifies root CLAUDE.md and adds a spec in `docs/specs/`, and the conventions reviewer flagged spec divergence (heuristic). The bot review is a complement to, not a replacement for, a human read here.

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- nit: CLAUDE.md "Existing examples" lists `brand-claims` but it is introduced by this PR - consider "Examples:" instead - `CLAUDE.md:248`
- suggestion: Add a test for the "all daily sheets, none on Monday" scenario to exercise the eligibility filter returning no eligible sheet from a non-empty listing - `test/audits/brand-claims.test.js:592`
- nit: Spec Technical Design steps 4-5 should mention the Monday-eligibility filter (LLMO-6877) and dynamic cadence detection to stay in sync with the implementation - `docs/specs/2026-08-11-brand-claims-scheduled-trigger.md:71`

</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 27s | <b>Cost:</b> $4.29 | <b>Commit:</b> <code>a336dbdd40c7d1d6dc7aba83e8264b9491098e9b</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
